src/main/java/uk/gov/ons/ctp/response/action/service/ActionProcessingService.java

```diff
@@ -35,66 +33,119 @@
   public static final String CREATED = "CREATED";
   public static final String ENABLED = "ENABLED";
   public static final String NOTIFY = "Notify";
+  public static final String PRINTER = "Printer";
   public static final String PENDING = "PENDING";
 
-  @Autowired private ActionRepository actionRepo;
-
-  @Autowired private ActionInstructionPublisher actionInstructionPublisher;
+  @Autowired private ActionCaseRepository actionCaseRepo;
 
   @Autowired private NotifyService notifyService;
 
-  @Autowired
-  private StateTransitionManager<ActionDTO.ActionState, ActionDTO.ActionEvent>
-      actionSvcStateTransitionManager;
+  @Autowired private NotificationFileCreator notificationFileCreator;
+
+  @Autowired private ActionStateService actionStateService;
 
-  private ActionRequestDecorator[] decorators;
+  @Autowired private ActionRequestContextFactory decoratorContextFactory;
 
-  public abstract ActionRequestContextFactory getActionRequestDecoratorContextFactory();
+  private static final ActionRequestDecorator[] DECORATORS = {
+    new ActionAndActionPlan(),
+    new CaseAndCaseEvent(),
+    new CollectionExerciseAndSurvey(),
+    new PartyAndContact(),
+    new SampleUnitRef()
+  };
 
-  public ActionProcessingService(ActionRequestDecorator[] decorators) {
-    this.decorators = decorators;
+  public void processActions(ActionType actionType, List<Action> allActions) {
+    if (actionType == null) {
+      throw new IllegalStateException(ACTION_TYPE_NOT_DEFINED);
+    }
+    String handler = actionType.getHandler();
+    if (handler == null) {
+      log.with("handler", actionType.getHandler()).error("no action type handler provided");
+    } else if (handler.equals(NOTIFY)) {
+      processEmails(actionType, allActions);
+    } else if (actionType.getHandler().equals(PRINTER)) {
+      processLetters(actionType, allActions);
+    } else {
+      log.with("handler", actionType.getHandler())
+          .with("actions", allActions.size())
+          .error("unsupported action type handler");
+    }
   }
 
-  /** Distributes requests for a single action */
-  @Transactional(propagation = Propagation.REQUIRES_NEW)
-  public void processActionRequests(final UUID actionId) {
-    Action action = actionRepo.findById(actionId);
-    log.with("actionId", action.getId()).info("Processing actionRequest");
+  public void processLetters(ActionType actionType, List<Action> allActions) {
+    // action requests are decorated actions
+    log.with("actions", allActions.size()).debug("processing letters");
+    List<ActionRequest> printerActions = new ArrayList<>();
+    for (Action action : allActions) {
+      if (checkCaseExists(action) && validate(action) && !cancelled(action)) {
```

Comment:
Do we need to log anything if any of these fail?  Even just a simple `"Not going to process this"`

Reply:
Done 👍